qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,337 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An implementation of the AdaptVQE algorithm."""
+from __future__ import annotations
+from typing import Optional, List, Tuple, Union
+
+import re
+import logging
+
+from enum import Enum
+import numpy as np
+from qiskit import QiskitError
+
+from qiskit.algorithms import VQE
+from qiskit.algorithms.list_or_dict import ListOrDict
+from qiskit.algorithms.minimum_eigen_solvers.vqe import VQEResult
+from qiskit.algorithms import VariationalAlgorithm
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import OperatorBase, PauliSumOp, CircuitSampler
+from qiskit.opflow.gradients import GradientBase, Gradient
+from qiskit.circuit.library import EvolvedOperatorAnsatz
+from qiskit.utils.validation import validate_min
+from qiskit.algorithms.aux_ops_evaluator import eval_observables
+from ..exceptions import AlgorithmError
+
+
+logger = logging.getLogger(__name__)
+
+
+class FinishingCriterion(Enum):
```

Comment:
We agreed with @mrossinek that `TerminationCriterion` would be a better name.

Reply:
When adapting the name of this class, it would be nice to change the variable name further down in the code accordingly: `finishing_criterion -> termination_criterion`